corgi/core/models.py

```diff
@@ -930,12 +932,27 @@ def latest_components(
         include: bool = True,
     ) -> "ComponentQuerySet":
         """Return components from latest builds across all product streams."""
-        distinct_components = (
-            self.values_list("namespace", "name", "arch").order_by().distinct().iterator()
+        nevras_by_components = self.values_list("namespace", "name", "arch").annotate(
+            nevras=JSONBAgg(
+                RawSQL(
+                    "json_build_object('nevra', nevra, 'epoch', epoch, 'version', core_component.version, 'release', release)",
+                    (),
+                )
+            )
```

Comment:
I will add a comment which explains what this is doing. This would be simpler and not require JSONBAgg or Raw SQL if we just aggregated the nevras. I didn't do that because we don't take the nevras from Brew, but build it ourselves from the name/epoch/version/release/arch values in the get_nevra function of Component. I'm worried we will build it incorrectly which will lead to bad results here. Also, we'd have to strip the name and arch values before using labelCompare because it expects only a tuple of epoch/version/release. Fetching each individual field allows us to ensure we have the correct value for each directly from Brew, and also not have to rebuild the nevra once we find the latest.

The reason we stopped using the database to do the sorting was that we need to use RPM schemantics to compare the versions, see this comment from [RHEL security architect](https://issues.redhat.com/browse/CORGI-391?focusedId=21407021&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-21407021). The rules around this are complicated and not something that's easy to replicate in the DB directly.

When I was originally testing this, I added core_component table name to version because the database seemed to be confused about which 'version' field we where after. However I tested removing the table name now and it didn't complain so I'm going to remove that.

Reply:
Actually I got the error again about version. So I'm going to qualify the table.

`psycopg2.errors.AmbiguousColumn: column reference "version" is ambiguous
LINE 1: ...object('nevra', nevra, 'epoch', epoch, 'version', version,`